src/carnot/exec/otel_export_sink_node.cc

```diff
@@ -90,6 +91,14 @@ Status OTelExportSinkNode::CloseImpl(ExecState* exec_state) {
   return Status::OK();
 }
 
+void SetStringOrBytes(std::string str, ::opentelemetry::proto::common::v1::KeyValue* otel_attr) {
+  if (!simdutf::validate_utf8(str.data(), str.length())) {
```

Comment:
I'm a little worried about the performance overhead of this check for every single byte of data that passes through, but at the same time I don't really see an alternative.